### DISCUSSION TITLE
the minimum value of S11

### DISCUSSION THREAD

**Discussion by ghost** [ansys/pyaedt#3836] [Q&A] [answered]
Hi,

I asked how to read the minimum value of S11 using PyAEDT in this community before.

At that time, I was told that I should refer to the folloing site:
https://aedt.docs.pyansys.com/version/stable/examples/02-HFSS/Probe_Fed_Patch.html#sphx-glr-examples-02-hfss-probe-fed-patch-py

I knew how to plot S11 referring to this site.
But, this data isn't a integer type, so it displays a error when I try to read the minimum value of this data using the min() function.

Can anyone tell me how to convert S11 data into numbers?
I would like to find the minimum value of S11.

Thank you.

**Comment by Samuelopez-ansys** [comment, marked as answer]
Hi @crapemyrtle-841 ,

I will use the same project you shared here: https://github.com/ansys/pyaedt/discussions/3817

The method get_solution_data returns an object, which contains all the needed information, and in addition, method to operate with the data in a very efficient way. Please check PyAEDT examples which can give you different ideas of the power of this method. But just to give you the solution of your question, please check the following code:


```
import pyaedt

pyaedt.settings.use_grpc_api = False

version = '2023.2'
print("Loading the PyAEDT Console.")
project = r"C:\Users\slopez\Downloads\antenna_test.aedt"
hfss = Hfss(projectname=project, designname="Design1", specified_version=version, new_desktop_session=False,
            non_graphical=False, close_on_exit=False, student_version=False)

hfss["L1"] = "26.5mm"

hfss.analyze("Setup1", num_cores=4)

# Get available S parameters
traces = hfss.post.available_report_quantities("S Parameter")

solution = hfss.post.get_solution_data(traces[0])

s11_db = solution.data_db20()
```

In the following code, once the project is solved, you can take any available trace, for example I am saving in a list called "traces"  all the available S parameters (in this case there is only one because you have 1 port). 

Then I get the data of this single traces with the method get_solution_data. It returns an object, for more information, please go to the API Documentation:
https://aedt.docs.pyansys.com/version/stable/API/_autosummary/pyaedt.modules.AdvancedPostProcessing.PostProcessor.get_solution_data.html


In this case, with a single line I took the data and I obtained the S11 dB20, you can compare it with AEDT results:

<img width="935" alt="image" src="https://github.com/ansys/pyaedt/assets/85613111/16f81f14-6c6f-483f-8d6a-31cde6f21ce2">

From here is quite easy to find the minimum:

`min(s11_db)`

And you can even find the resonance frequency:

```
freq_index = s11_db.index(min(s11_db))
freqs = solution.primary_sweep_values
resonance = freqs[freq_index]
```

Finally, you can even perform a simple plot using matplotlib:

`solution.plot()
`
<img width="833" alt="image" src="https://github.com/ansys/pyaedt/assets/85613111/bec8e8c3-3e7a-45d4-bd23-ea78ede44eab">

I think you will agree this is a really powerful tool :)

**Reply by ghost** [reply]
Hi @Samuelopez-ansys 

It was easy to understand. Thank you for teaching me so kindly.

The code you taught me solved the problem.
I tried to look it up, but I couldn't find it and really needed someone's help.

I truly appreciate that. Thank you so much!
